cmake/environmentchecks.cmake

```diff
@@ -108,12 +107,33 @@ set(HAVE_LIBXML2_CONFIG FALSE)
 # need to track how we found LibXml2.
 find_package(LibXml2 CONFIG QUIET)
 if(LibXml2_FOUND)
-  if(TARGET zlib)
+  if(TARGET z)
     set(HAVE_ZLIB_TARGET TRUE)
+    get_target_property(ZLIB_TARGET_TYPE z TYPE)
   else()
-    find_package(ZLIB REQUIRED)
+    find_package(ZLIB CONFIG QUIET)
+    if(ZLIB_FOUND)
+      if(TARGET z)
+        set(HAVE_ZLIB_TARGET TRUE)
+        get_target_property(ZLIB_TARGET_TYPE z TYPE)
+      endif()
+    endif()
+    if(NOT ZLIB_FOUND)
+      find_package(ZLIB REQUIRED)
+    endif()
   endif()
   set(HAVE_LIBXML2_CONFIG TRUE)
+  # Different versions of LibXml2 have different names for the library target.
+  # We try and capture that here.
+  if(TARGET xml2)
+    set(LIBXML2_TARGET_NAME xml2)
+  elseif(TARGET LibXml2)
+    set(LIBXML2_TARGET_NAME LibXml2)
+  else()
+    message(FATAL_ERROR "FindLibXml2: Found config for LibXml2 but could not determine target name.")
```

Comment:
I would just output `Found configuration for libxml2 but could not determine its target name.`

Reply:
This is more consistent with other CMake error messages from find package errors.

Reply:
Ok, I am not going to insist. All I can say is that I had a quick look at the fatal error messages we generate and I couldn't find any kind of consistency.